aim/api/resource.py

```diff
@@ -291,6 +291,225 @@ def __init__(self, **kwargs):
+
+
+class SpanVsource(AciResourceBase):
+    """Resource representing a ERSPAN VSource in ACI.
+
+    Identity attributes are RNs for ERSPAN VSource.
+    """
+
+    identity_attributes = t.identity(
+        ('vsg_name', t.name),
+        ('name', t.name))
+    other_attributes = t.other(
+        ('dir', t.enum("in", "out", "both")),
+        ('monitored', t.bool),
+        ('display_name', t.name))
+
+    _aci_mo_name = 'spanVSrc'
+    _tree_parent = SpanVsourceGroup
+
+    def __init__(self, **kwargs):
+        super(SpanVsource, self).__init__({'dir': 'both',
+                                           'monitored': False},
+                                          **kwargs)
+
+
+class SpanSrcVport(AciResourceBase):
+    """Resource representing a Relation to SPAN Source to Port.
+
+    Identity attributes are name of Source Group name, source name
+    and src_path of the VSPAN session.
+    """
+
+    identity_attributes = t.identity(
+        ('vsg_name', t.name),
+        ('vs_name', t.name),
+        ('src_path', t.string()))
+    other_attributes = t.other(
+        ('monitored', t.bool))
+
+    _aci_mo_name = 'spanRsSrcToVPort'
+    _tree_parent = SpanVsource
+
+    def __init__(self, **kwargs):
+        super(SpanSrcVport, self).__init__({'monitored': False},
+                                           **kwargs)
+
+
+class SpanVdestGroup(AciResourceBase):
+    """Resource representing a ERSPAN VDest Group in ACI.
+
+    Identity attributes are RNs for ERSPAN VDest Group.
+    """
+
+    identity_attributes = t.identity(
+        ('name', t.name))
+    other_attributes = t.other(
+        ('monitored', t.bool),
+        ('display_name', t.name))
+
+    _aci_mo_name = 'spanVDestGrp'
+    _tree_parent = Infra
+
+    def __init__(self, **kwargs):
+        super(SpanVdestGroup, self).__init__({'monitored': False},
+                                             **kwargs)
+
+
+class SpanVdest(AciResourceBase):
+    """Resource representing a ERSPAN VDest in ACI.
+
+    Identity attributes are RNs for ERSPAN VDest.
+    """
+
+    identity_attributes = t.identity(
+        ('vdg_name', t.name),
+        ('name', t.name))
+    other_attributes = t.other(
+        ('monitored', t.bool),
+        ('display_name', t.name))
+
+    _aci_mo_name = 'spanVDest'
+    _tree_parent = SpanVdestGroup
+
+    def __init__(self, **kwargs):
+        super(SpanVdest, self).__init__({'monitored': False},
+                                        **kwargs)
+
+
+class SpanVepgSummary(AciResourceBase):
+    """Resource representing a ERSPAN Destination VEpg Summary in ACI.
+
+    Identity attributes are RNs for ERSPAN Destination VEpg Summary.
+    """
+
+    identity_attributes = t.identity(
+        ('vdg_name', t.name),
+        ('vd_name', t.name))
+    other_attributes = t.other(
+        ('monitored', t.bool),
+        ('display_name', t.name),
+        ('dst_ip', t.string()),
+        ('flow_id', t.string()),
+        ('ttl', t.string()),
```

Comment:
we should use t.integer for ttl and mtu field I think.

Reply:
I tried using t.integer. But got a few errors wrt integer out of scope. hence switched back to string.

Reply:
hmm it should work. Can you try to debug to see where that error is coming from?

Reply:
will do